x-pack/plugins/enterprise_search/public/applications/app_search/components/api_logs/api_logs_logic.ts

```diff
@@ -122,9 +122,14 @@ export const ApiLogsLogic = kea<MakeLogicType<ApiLogsValues, ApiLogsActions>>({
         }
       } catch (e) {
         if (isPoll) {
-          // If polling fails, it will typically be due due to http connection -
+          // If polling fails, it will typically be due to http connection -
           // we should send a more human-readable message if so
-          setErrorMessage(POLLING_ERROR_MESSAGE);
+          flashErrorToast({
+            id: 'apiLogsPollError',
+            title: POLLING_ERROR_TITLE,
+            text: POLLING_ERROR_TEXT,
+            toastLifeTimeMs: POLLING_DURATION * 0.75,
```

Comment:
This is relevant to https://github.com/elastic/kibana/pull/95981#discussion_r605235539 - linking because @scottybollinger also a left a comment there on the topic:

> With the proposed limit of 5 seconds, there aren't any real cases where we'll have too many stacked toasts, with a few exceptions like the API logs errors. We could try to enforce a max num of toasts at any given time, but I am not sure that adds any value.

I think my only hesitation about changing the `addToastMessage` to `setToastMessages` is that the worst case scenario for `setToastMessages` is worse than `addToastMessage`.

- `addToastMessage` - worst case scenario is a stacked toasts, which EUI recommends against as a guideline but doesn't explicitly disallow
- `setToastMessages` - worst case scenario is we call this too many times in a row **and previous unique messsages are completely swallowed**/no longer visible to users

To be fair, that same issue is potentially also there for our setFlashMessages helpers, so I could be overthinking it - but I _think_ I lean towards potentially having issues with over-messaging rather than under-messaging as a worst case scenario.

For this specific scenario, I could remove this custom `toastLifeTimeMs` from this toast obj and the behavior would still work fine - the `0.75` reduction is more of a UI/UX nice to have, and the # of polling-based events we have in the codebase is currently fairly low, so I'm not overly worried about it.